### ISSUE TITLE
Proposal: jiant.simple

### ISSUE THREAD

**Issue by zphang** [jiant-dev/jiant#21]
As a radical departure from the current development path, I want to propose a `jiant.simple` API that goes in a completely different direction: 

* Tons of sensible defaults, built for in-session/notebook usage (e.g. targeting student usage).
* Built on top of the regular `jiant` components
* No expectation of feature parity with broader jiant
* Ideally components are modifiable with more complex `jiant` components, but even this can be relaxed.

A brief sketch of usage looks like this:

### Setup Tasks
```python
import jiant.simple  # a single import
mnli_task = jiant.simple.create_task({
    name="mnli"
    task_template="TwoInputClassification",
    task_args={
        "num_labels": 3,
        "labels_list": ["entailment", "contradiction", "neutral"],
    }
    evaluation=jiant.simple.evaluation.Accuracy,
})
mnli_train_examples = [
    mnli_task.example({"text_a": "hi", "text_b": "hello", label="yes"}),
]
mnli_val_examples = [
    mnli_task.example({"text_a": "hi", "text_b": "hello", label="no"}),
]
cola_task = jiant.simple.create_task({
    name="cola"
    task_template="OneInputClassification",
    task_args={
        "num_labels": 2,
        "labels_list": [False, True],
    }
    evaluation=jiant.simple.evaluation.Accuracy,
})
cola_train_examples = [
    cola_task.make_example({"text": "hi", label=False}),
]
```

### Model and runner
```python
jiant_model = jiant.simple.create_model(
    model_type="transformers:roberta-large",
    load_pretrained=True,
)
jiant_model.add_head_for_task(task=mnli_task)
jiant_model.add_head_for_task(task=cola_task)
jiant_optimizer_scheduler = jiant.simple.create_optimizer_scheduler(
    model=model,
    optimizer="AdamW",
    lr=1e-5,
    schedule="LinearWith10%Warmup",
)
jiant_runner = jiant.simple.create_runner(
    model=jiant_model,
    optimizer_scheduler=jiant_optimizer_scheduler,
    task_data=[
        {
            "task": mnli_task,
            "train": mnli_train_examples,
            "val": mnli_val_examples,
        },
        {
            "task": cola_task,
            "train": cola_train_examples,
        }
    ],
    val_tasks=["mnli"],
    sampling="proportional",
)
```

### Training
```python
best_metrics = jiant.simple.InitialMetric()
best_state_dict = jiant_runner.get_model_state_dict()
jiant_runner.set_max_steps(10000)
for i in range(10):
    jiant_runner.train(steps=1000)
    val_metrics = runner.val(subset=100)
    if val_metrics > best_metrics:
        best_state_dict = jiant_runner.get_model_state_dict()
jiant_runner.model.load_state_dict(best_state_dict)
```

### Evaluation
```python
val_results = runner.run_val()

# I decided I also want to evaluate on XNLI-French (zero-shot)
xnli_fr_task = jiant.simple.clone_task(
    task=mnli_task,
    name="xnli_fr",
)
xnli_fr_val_examples = [
    xnli_fr_task.make_example({"text_a": "bonjour", "text_b": "bonsoir", label=False}),
]
xnli_result = jiant_runner.run_val(
    task={
        "task": xnli_fr_task,
        "using": "mnli",
        "val" xnli_fr_val_examples,
    }
)
```

**Comment by sleepinyourhat** [comment]
Could you say more about how this would improve upon the 'fork a HuggingFace example' workflow that seems to be the standard for that kind of use case right now?

**Comment by sleepinyourhat** [comment]
(I'm interested, though, if this is something that legitimately takes advantage of our infrastructure; rather than just being a separate thin wrapper.)

**Comment by zphang** [comment]
I feel like the question about our value-add on top of 'fork a HuggingFace example' becomes harder to answer everyday...

That said:

* Multi-task models and the encoders+heads structure continue to be a primary/defining feature of jiant
* The Simple API will build directly on `jiant` components, and mostly put them together in a standard-ish workflow. It should generally be possible to mix and match "Simple" and "Full" jiant components.
* This will also be a good starting point for beginners to get introduced to the `jiant` code and components, without getting overwhelmed by options
